Orange/data/table/base.py

```diff
@@ -0,0 +1,1292 @@
+                    self[var.name] = self[var.name].apply(lambda v: var.values[int(v)])
+
+    def _transform_discrete_into_categorical(self):
+        """Transform columns with discrete variables into pandas' categoricals.
+
+        The operation is performed in-place.
+        This must be done after replacing null values because those aren't values,
+        and also after transforming discretes to descriptors.
+        """
+        for var in chain(self.domain.variables, self.domain.metas):
+            if isinstance(var, DiscreteVariable):
+                self[var.name] = pd.Categorical(self[var.name], categories=var.values, ordered=var.ordered)
+
+    def _transform_timevariable_into_datetime(self):
+        """Transform columns with TimeVariables into pandas' datetime columns (in-place)."""
+        for var in chain(self.domain.variables, self.domain.metas):
+            if isinstance(var, TimeVariable):
+                self[var.name] = var.column_to_datetime(self[var.name])
+
+    def save(self, filename):
+        """Save a Table to a file.
+
+        Parameters
+        ----------
+        filename : str
+            The destination file path, absolute or relative.
+        """
+        ext = os.path.splitext(filename)[1]
+        from Orange.data.io import FileFormat
+        writer = FileFormat.writers.get(ext)
+        if not writer:
+            desc = FileFormat.names.get(ext)
+            if desc:
+                raise IOError("Writing of {}s is not supported".format(desc.lower()))
+            else:
+                raise IOError("Unknown file name extension.")
+        writer.write_file(filename, self)
+
+    def __getitem__(self, item):
+        # if selecting a column subset, we need to transfer weights so they don't just disappear
+        # only do this for multiple column selection, which returns a DataFrame by contract
+        if isinstance(item, (Sequence, pd.Index)) and not isinstance(item, str) \
+                and all(isinstance(i, str) for i in item) \
+                and self._WEIGHTS_COLUMN not in item:
+            item = list(item) + self._INTERNAL_COLUMN_NAMES
+        return super().__getitem__(item)
+
+    def __setitem__(self, key, value):
+        # if the table has an empty index and we're inserting a new row,
+        # the index would be created by pandas automatically.
+        # we want to maintain unique indices, so we override the index manually.
+        # we also need to set default weights, lest they be NA
+        new_index_and_weights = len(self.index) == 0
+
+        # PANDAS CONTRACT BREAKAGE:
+        # the pandas default behaviour when adding a new column as a Series is that
+        # the indexes are inner-joined: only the elements at the indices that exist
+        # in the current table are actually set, other elements are NA
+        # for easier handling of Orange behaviour, we effectively ignore the index
+        # on the series to just merge the column into the table
```

Comment:
Because what I was fearing is that someone dare use `.reindex()` (e.g. on some datetime) and then rely on pandas to auto do the right (standard, documented) stuff when some other series with a datetime index is attached. This should work, right?